### PR TITLE
[DSCP-152] Wallet backend impementation

### PR DESCRIPTION
**Pull request by Martoon-00**
Implements simple wallet backend.

- [x] Implement backend
- [x] Add servant server for backend
- [x] Add servant client requests
- [x] Copy-paste code of servant requests logging
- [ ] ~Add mempool support~ separate PR
- [ ] ~Show tx list in account state~ in unforeseen future

### CODE REVIEW

**Comment by goodsoft** [comment]
I thought a bit about point 1, and I think the server part would fit perfectly in `Dscp.Web.Wallet` (instead of current `Dscp.Wallet.Web`)

**Comment by Martoon-00** [comment]
Could you explain such choice?

**Comment by goodsoft** [comment]
Because if there are multiple HTTP-endpoints in Disciplina, they could all fit under `Dscp.Web` namespace along with all required `Types` and all.

**Comment by Martoon-00** [comment]
Both points fixed

File: witness/src/Dscp/Witness/Web/Wallet/API.hs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import Servant.Generic
+
+import Dscp.Core.Types
+import Dscp.Witness.Web.Wallet.Types
+
+-- | API/implementation of wallet endpoints.
+data WalletEndpoints route = WalletEndpoints
+    { -- shortened prefix for purpose, one will most probably use type aliases
+      -- refering this type
+      wGetNextNonce :: route
```
Comment (volhovm) [inline witness/src/Dscp/Witness/Web/Wallet/API.hs]: Why not return the nonce with the account state?

Comment (Martoon-00) [reply witness/src/Dscp/Witness/Web/Wallet/API.hs]: Oki, fixed

**Pull request merged**